### PR TITLE
fix: remove Firefox icon border when using animate-spin

### PR DESCRIPTION
**Pull request by userquin**
https://streamable.com/og19l7

Fix here: https://github.com/unocss/unocss/issues/300#issuecomment-1000886775

### CODE REVIEW

**Review by edimitchel — APPROVED** [review]
Great music (and great fix!)

File: components/notification/NotificationEnablePushNotification.client.vue [hunk 1]
```diff
@@ -66,7 +66,10 @@ const isLegacyAccount = computed(() => !currentUser.value?.vapidKey)
       :disabled="busy || isLegacyAccount"
       @click="$emit('subscribe')"
     >
-      <span aria-hidden="true" :class="busy && animate ? 'i-ri:loader-2-fill animate-spin' : 'i-ri:check-line'" />
+      <span v-if="busy && animate" aria-hidden="true" block animate-spin preserve-3d>
```
Comment (sxzz) [inline components/notification/NotificationEnablePushNotification.client.vue:69]:
```suggestion
      <div v-if="busy && animate" aria-hidden="true" animate-spin preserve-3d>
```

Same for belows

File: components/notification/NotificationPreferences.client.vue [hunk 2]
```diff
@@ -147,7 +147,10 @@ onActivated(() => (busy = false))
                   :class="busy || !saveEnabled ? 'border-transparent' : null"
                   :disabled="busy || !saveEnabled"
                 >
-                  <span :class="busy && animateSave ? 'i-ri:loader-2-fill animate-spin' : 'i-ri:save-2-fill'" />
+                  <span v-if="true" aria-hidden="true" block animate-spin preserve-3d>
```
Comment (sxzz) [inline components/notification/NotificationPreferences.client.vue:150]: debugging? We should revert it.

Comment (userquin) [reply components/notification/NotificationPreferences.client.vue:150]: upps

Comment (userquin) [reply components/notification/NotificationPreferences.client.vue]: done

File: components/notification/NotificationEnablePushNotification.client.vue [hunk 3]
```diff
@@ -66,7 +66,10 @@ const isLegacyAccount = computed(() => !currentUser.value?.vapidKey)
       :disabled="busy || isLegacyAccount"
       @click="$emit('subscribe')"
     >
-      <span aria-hidden="true" :class="busy && animate ? 'i-ri:loader-2-fill animate-spin' : 'i-ri:check-line'" />
+      <div v-if="busy && animate" aria-hidden="true" animate-spin preserve-3d>
+        <div block i-ri:loader-2-fill aria-hidden="true" />
```
Comment (userquin) [inline components/notification/NotificationEnablePushNotification.client.vue:70]: arrgg

File: components/notification/NotificationEnablePushNotification.client.vue [hunk 4]
```diff
@@ -66,7 +66,10 @@ const isLegacyAccount = computed(() => !currentUser.value?.vapidKey)
       :disabled="busy || isLegacyAccount"
       @click="$emit('subscribe')"
     >
-      <span aria-hidden="true" :class="busy && animate ? 'i-ri:loader-2-fill animate-spin' : 'i-ri:check-line'" />
+      <div v-if="busy && animate" aria-hidden="true" animate-spin preserve-3d>
+        <div i-ri:loader-2-fill aria-hidden="true" />
+      </div>
+      <div v-else aria-hidden="true" i-ri:check-line />
```
Comment (danielroe) [inline components/notification/NotificationEnablePushNotification.client.vue:72]: would you update here and elsewhere so we're not change the elements from span -> div in this PR?

File: components/notification/NotificationPreferences.client.vue [hunk 5]
```diff
@@ -147,7 +147,10 @@ onActivated(() => (busy = false))
                   :class="busy || !saveEnabled ? 'border-transparent' : null"
                   :disabled="busy || !saveEnabled"
                 >
-                  <span :class="busy && animateSave ? 'i-ri:loader-2-fill animate-spin' : 'i-ri:save-2-fill'" />
+                  <div v-if="true" aria-hidden="true" animate-spin preserve-3d>
```
Comment (sxzz) [inline components/notification/NotificationPreferences.client.vue:150]: There's another place should be reverted too.

File: components/notification/NotificationEnablePushNotification.client.vue (hunk 4 above)
Comment (userquin) [reply components/notification/NotificationEnablePushNotification.client.vue:72]: @danielroe @sxzz I'll keep original elements updating only span that requires block: the icon not being displayed when using span inside span.

Comment (danielroe) [reply components/notification/NotificationEnablePushNotification.client.vue:72]: can we find another solution for that? we shouldn't have any `<div>` inside `<button>` - it's not permitted HTML content

Comment (userquin) [reply components/notification/NotificationEnablePushNotification.client.vue:72]: yeah, I get this warning using IntelliJ, all my components using span inside button

Comment (userquin) [reply components/notification/NotificationEnablePushNotification.client.vue:72]: I'll revert last commit, you need to review a few components...

File: components/common/CommonInputImage.vue [hunk 6]
```diff
@@ -98,7 +98,9 @@ watch(file, (image, _, onCleanup) => {
       bg="black/30" text-white
       flex justify-center items-center
     >
-      <div class="i-ri:loader-4-line animate-spin animate-duration-[2.5s]" text-4xl />
+      <div class="animate-spin animate-duration-[2.5s] preserve-3d">
```
Comment (userquin) [inline components/common/CommonInputImage.vue:101]: this is also not allowed

Comment (userquin) [reply components/common/CommonInputImage.vue:101]: same for divs inside buttons:

imagen

Comment (danielroe) [reply components/common/CommonInputImage.vue:101]: quite right. those should also be converted to spans.

File: components/notification/NotificationEnablePushNotification.client.vue (hunk 4 above)
Comment (userquin) [reply components/notification/NotificationEnablePushNotification.client.vue]: done

File: components/common/CommonInputImage.vue (hunk 6 above)
Comment (userquin) [reply components/common/CommonInputImage.vue:101]: fixing label...

Comment (userquin) [reply components/common/CommonInputImage.vue]: done

File: components/notification/NotificationPreferences.client.vue (hunk 2 above)
Comment (edimitchel) [inline components/notification/NotificationPreferences.client.vue:150]: Still `v-if="true"`

**Review by edimitchel — CHANGES_REQUESTED** [review]

File: components/notification/NotificationPreferences.client.vue (hunk 2 above)
Comment (userquin) [reply components/notification/NotificationPreferences.client.vue]: done

**Review by edimitchel — COMMENTED** [review]
LGTM

**Review by edimitchel — APPROVED** [review]

**Review by sxzz — APPROVED** [review]

**Pull request merged**